src/server/scripts/Shadowlands/zone_exiles_reach.cpp

```diff
@@ -0,0 +1,546 @@
+{
+public:
+    conversation_horde_sparing_partner() : ConversationScript("conversation_horde_sparing_partner") { }
+
+    void OnConversationCreate(Conversation* conversation, Unit* creator) override
+    {
+        std::list<Creature*> spar;
+        creator->GetCreatureListWithEntryInGrid(spar, NPC_SPARING_PARTNER, 25.0f);
+        for (Creature* creature : spar)
+        {
+            if (creature->GetDemonCreatorGUID() == creator->GetGUID()) // @TODO requires research about DemonCreator usage for summons to work
+                conversation->AddActor(creature->GetEntry(), 1, creature->GetGUID());
+        }
+    }
+};
+
+enum PlayerScriptHordeShipCrash
+{
+    MOVIE_ALLIANCE_SHIP_CRASH = 895,
+    MOVIE_HORDE_SHIP_CRASH    = 931,
+    SPELL_ALLIANCE_SHIP_CRASH = 305446,
+    SPELL_HORDE_SHIP_CRASH    = 325133
+};
+
+class player_ship_crash : public PlayerScript
+{
+public:
+    player_ship_crash() : PlayerScript("player_ship_crash") { }
+
+    void OnMovieComplete(Player* player, uint32 movieId) override
+    {
+        switch (movieId)
+        {
+            case MOVIE_ALLIANCE_SHIP_CRASH:
+                player->CastSpell(player, SPELL_ALLIANCE_SHIP_CRASH, true);
+                break;
+            case MOVIE_HORDE_SHIP_CRASH:
+                player->CastSpell(player, SPELL_HORDE_SHIP_CRASH, true);
+                break;
+            default:
+                break;
+        }
+    }
+};
+
+enum AllianceHordeShipSceneSpells
+{
+    SPELL_BEGIN_TUTORIAL = 295600,
+    SPELL_KNOCKED_DOWN   = 305445,
+    SPELL_CRASHED_LANDED = 325136
+};
+
+class scene_alliance_and_horde_ship : public SceneScript
+{
+public:
+    scene_alliance_and_horde_ship() : SceneScript("scene_alliance_and_horde_ship") { }
+
+    void OnSceneComplete(Player* player, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/) override
+    {
+        player->CastSpell(player, SPELL_BEGIN_TUTORIAL, true);
+    }
+
+    void OnSceneCancel(Player* player, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/) override
+    {
+        player->CastSpell(player, SPELL_BEGIN_TUTORIAL, true);
+    }
+};
+
+class scene_alliance_and_horde_crash : public SceneScript
+{
+public:
+    scene_alliance_and_horde_crash() : SceneScript("scene_alliance_and_horde_crash") { }
+
+    void OnSceneTriggerEvent(Player* player, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/, std::string const& triggerName) override
+    {
+        if (triggerName == "Begin Knockdown Aura")
+            player->CastSpell(player, SPELL_KNOCKED_DOWN, true);
+    }
+
+    void OnSceneComplete(Player* player, uint32 /*sceneInstanceID*/, SceneTemplate const* /*sceneTemplate*/) override
+    {
+        player->CastSpell(player, SPELL_CRASHED_LANDED, true);
+    }
+};
+
+class spell_summon_throg_combat_training : public SpellScript
+{
+    PrepareSpellScript(spell_summon_throg_combat_training);
+
+    void RelocateTransportOffset(SpellEffIndex /*effIndex*/)
+    {
```

Comment:
Remove the script if you don’t need it anymore ?

Reply:
(You shouldn't mark a comment as resolved which you didn't resolve)

Remove the script if you don’t need it anymore ?